ethereumj-core/src/main/java/org/ethereum/net/eth/handler/EthHandler.java

```diff
@@ -99,6 +104,10 @@ public void channelRead0(final ChannelHandlerContext ctx, EthMessage msg) throws
         msgQueue.receivedMessage(msg);
     }
 
+    public Publisher getPublisher() {
+        return ((BackwardCompatibilityEthereumListenerProxy) ethereumListener).getPublisher();
```

Comment:
I think `BackwardCompatibilityEthereumListenerProxy` should be localized somehow and don't spread to such classes.

Reply:
The main idea of `BackwardCompatibilityEthereumListenerProxy` is to encapsulate new and old publishing mechanisms under` EthereumListener` interface, and proxy old messages to both of them with / without transformation. We should use this component everywhere to reduce the number of changes and support both mechanisms at the same time.
`BackwardCompatibilityEthereumListenerProxy` will simplify removing deprecated logic in the future. And when the time comes we'll remove `BackwardCompatibilityEthereumListenerProxy` and all` EthereumListener` related stuff and leave only `Publisher` invokes.

Reply:
I got this idea as we are moving everything inside our application to the new subscribe-publish model, while leaving common user interfaces and marking them as deprecated. So any backward compatibility gateways should be localized to old public user interfaces.